src/farcasterd/runtime.rs

```diff
@@ -1337,8 +509,8 @@ impl Runtime {
             }
 
             // Returns a unique response that contains the complete progress queue
-            Request::ReadProgress(swapid) => {
```

Comment:
since cleaning up redundant state anyway, this `checkpointed_pub_offers` is quite unwieldy. Seems it's only needed for the trade state machine to get the public offer & trade role when fed a swap id.

Lightly suggest moving the public offer & trade role into the event msg passed to `transition_to_restoring_swapd` as well to get rid of this.